Modules/posixmodule.c

```diff
@@ -1576,9 +1571,10 @@ win32_xstat_impl(const wchar_t *path, struct _Py_stat_struct *result,
             return -1;
         }
         if (info.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) {
-            if (!win32_get_reparse_tag(hFile, &reparse_tag))
+            if (!win32_get_reparse_tag(hFile, &reparse_tag)) {
+                CloseHandle(hFile);
```

Comment:
This handle leak is still an issue, as well as the ones related to calling `get_target_path`. Can you update this PR to remove the `_getfinalpathname` parts that are no longer necessary?

Reply:
It looks like this PR was obsoleted by 3b20d3454e8082e07dba93617793de5dc9237828, which fixes the handle leak in this function as a side effect of the restructuring done for calling the function in a while loop and pre-initializing the memory.

The only functional difference is that this PR includes a regression test that #6010 doesn't.

Reply:
Your PR also identified and fixed leaks in `win32_xstat_impl` and `get_target_path`. I changed the title of issue 29734 to reflect this, but if you prefer I can close it and create a new issue for those leaks.

Reply:
The git diff is foils me once again. Completely forgot that I also looked at the other functions. :smile:

I think it's appropriate to keep this work under 29734, since its basically the same issue. For these changes, given the current state of the master branch, it may be easier to just close this PR and start a fresh branch, rather than dealing with the rebase. If I do, I'll make sure to tag you on the new PR so you're tracking it.

Either way, I'll also update the unit test to make calls to the other two functions so we can catch any regressions.